pkg/controller/issueapi/issue.go

```diff
@@ -112,21 +116,45 @@ func (c *Controller) IssueMany(ctx context.Context, requests []*IssueRequestInte
 		return results
 	}
 
-	if smsProvider != nil {
-		var wg sync.WaitGroup
-		for i, result := range results {
-			if result.ErrorReturn != nil {
-				continue
+	var wg sync.WaitGroup
+	for i, result := range results {
+		// Do not attempt to process things that have already errored.
+		if result.ErrorReturn != nil {
+			continue
+		}
+
+		// Get the associated request for this result.
+		issueReq := requests[i].IssueRequest
+
+		// Do not attempt to process requests that do not have a phone number.
+		if issueReq.Phone == "" {
```

Comment:
I think there's a missing error condition here where the phone number isn't provided BUT `OnlyGenerateSMS` is set to true

Reply:
All the error cases are [handled in validate_code.go](https://github.com/google/exposure-notifications-verification-server/pull/2081/files#diff-2187f7760dcfc954d7452c4188c16f6c208e47ed70865ebd7bce576d2e7e34faR76), which is called above when the results list is created.